containers/python/rpc.py

```diff
@@ -119,12 +127,20 @@ def handle_prediction_request(self, prediction_request):
             predict response
         """
         predict_fn = self.get_prediction_function()
-        # outputs = []
         total_length = 0
         outputs = predict_fn(prediction_request.inputs)
+        # Type check the outputs:
+        if not type(outputs) == list:
+            raise PredictionError("Model did not return a list")
+        if not type(outputs[0]) == str:
```

Comment:
It's possible that we'll get an empty list of outputs and crash due to an out of bounds list access here. We should swap this with the length check in the execution order.